packages/react-components/components/WithTextInputPasteAware.tsx

```diff
@@ -0,0 +1,85 @@
+// HOC to add a paste button to a text input
+
+import TouchableDefault from '@celo/react-components/components/Touchable'
+import Copy from '@celo/react-components/icons/Copy'
+import colors from '@celo/react-components/styles/colors'
+import * as React from 'react'
+import { AppState, Clipboard, StyleSheet, TextInputProps, View } from 'react-native'
+
+interface PasteAwareProps {
+  value: string
+  shouldShowClipboard: (value: string) => boolean
+  onChangeText: (text: string) => void
+}
+
+export default function withTextInputPasteAware<P extends TextInputProps>(
+  WrappedTextInput: React.ComponentType<P>
+) {
+  return class WithTextInputLabeling extends React.Component<P & PasteAwareProps> {
+    state = {
+      isPasteIconVisible: false,
+    }
+
+    async componentDidMount() {
+      AppState.addEventListener('change', this.checkClipboardContents)
+      await this.checkClipboardContents()
+    }
+
+    componentWillUnmount() {
+      AppState.removeEventListener('change', this.checkClipboardContents)
+    }
+
+    checkClipboardContents = async () => {
+      try {
+        const clipboardContent = await Clipboard.getString()
+        if (
+          clipboardContent &&
+          clipboardContent !== this.props.value &&
+          this.props.shouldShowClipboard(clipboardContent)
+        ) {
+          this.setState({ isPasteIconVisible: true })
+        } else {
+          this.setState({ isPasteIconVisible: false })
```

Comment:
nit: not really a problem here but this style of async usage could make it call `setState` after the component has been unmounted

Reply:
Is there a general strategy on how to avoid that?